README.md

```diff
@@ -17,41 +17,91 @@ Please visit the [latest stable release](https://github.com/Open-CMSIS-Pack/cpac
 ```bash
-$ cpackget pack|pdsc add|rm <pack-path>
-
-Options:
-
-  -h, --help        Help for cpackget.
-  -V, --version     Outout the version number of cpackget and exit.
-  -v, --verbose     Set verbosiness: None (Errors only), -v (Info) and -vv (Debugging)
+Usage:
+  cpackget [command] [flags]
+
+Available Commands:
+  help        Help about any command
+  index       Updates public index
+  init        Initializes a pack root folder
+  pack        Adds/Removes Open-CMSIS-Pack packages
+  pdsc        Adds or removes Open-CMSIS-Pack packages in the local file system via PDSC files.
+
+Flags:
+  -h, --help               help for cpackget
+  -R, --pack-root string   Specifies pack root folder. Defaults to CMSIS_PACK_ROOT environment variable
+  -q, --quiet              Run cpackget silently, printing only error messages
+  -v, --verbose            Sets verboseness level: None (Errors + Info + Warnings), -v (all + Debugging). Specify "-q" for no messages
+  -V, --version            Prints the version number of cpackget and exit
 
 Use "cpackget [command] --help" for more information about a command.
 ```
 
 For example, if one wanted help removing a pack, running `cpackget pack rm --help` would print out useful information on the subject.
 
-### Examples
+
+### Sepecifying the working pack root folder
+
+If cpackget is going to work on an existing pack root folder, there are two ways to specify it:
+
+1. `export CMSIS_PACK_ROOT=path/to/pack-root; cpackget pack add ARM.CMSIS`
+2. `cpackget --pack-root path/to/pack-root pack add ARM.CMSIS`
+
+The first example is more common because exporting the environment variables can happen only once.
+
+But cpackget is also capable of creating a new pack root folder if needed. For example:
+
+```
+$ cpackget init --pack-root path/to/new/pack-root https://keil.com/pack/index.pidx
+```
+
+The command will create a folder called `path/to/new/pack-root`, a few subfolders (`.Download`, `.Local`, `.Web`)
+and will place a copy of the index file (if specified) to `.Web/index.pidx`.
+
+If later it is needed to update the public index file, just run `cpackget index https://vendor.com/index.pidx` and
+`.Web/index.pidx` will be updated accordingly.
+
+
+### Adding packs
 
 The commands below are examples on how to add packs:
-- `cpackget pack add path/to/Vendor.PackName.x.y.z.pack`
-- `cpackget pack add https://vendor.com/example/Vendor.PackName.x.y.z.pack`
 
-If the pack is specified by an URL, cpackget command will download `Vendor.PackName.x.y.z.pack` before installing it under the **Open-CMSIS-Pack** root folder.
-This folder should be specified via system environment variable named **CMSIS_PACK_ROOT**. If empty, cpackget will attempt creating a folder called `.cpackget` in the local directory and that will become the installation folder.
+* `cpackget pack add path/to/Vendor.PackName.x.y.z.pack`
+* `cpackget pack add https://vendor.com/example/Vendor.PackName.x.y.z.pack`
+* `cpackget pack add Vendor.PackName.x.y.z`
```

Comment:
Install a pack version from the public package index. The download url will be looked up by the tool.

Reply:
ack